contracts/contracts/vault/VaultCore.sol

```diff
@@ -97,6 +107,46 @@ contract VaultCore is VaultStorage {
         }
     }
 
+    /**
+     * @dev Mint OUSD for OUSD Meta Strategy
+     * @param _amount Amount of the asset being deposited
+     *
+     * Notice: can't use nonReentrant modifier since BaseCurveStrategy's deposit
+     * already has that modifier present
+     */
+    function mintForStrategy(uint256 _amount)
+        external
+        whenNotCapitalPaused
+        onlyOusdMetaStrategy
+    {
+        require(_amount > 0, "Amount must be greater than 0");
+
+        emit Mint(msg.sender, _amount);
+
+        // Rebase must happen before any transfers occur.
+        // TODO: double check the relevance of this
+        if (_amount >= rebaseThreshold && !rebasePaused) {
+            _rebase();
+        }
+
+        netOusdMintedForStrategy += int256(_amount);
```

Comment:
Is this an unchecked conversion to int256()? If so, you might be able to pass in a very large mint of OUSD and have it get converted to a very small negative number, passing the require.

Reply:
Have done a check that amount value isn't too high. Will also add a test for that

Reply:
test added here: https://github.com/OriginProtocol/origin-dollar/pull/1002/commits/976868115c511f5a510f0a6a8ecfb4aed9f2e3c3